docs/introduction/basics.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: page
+title: "The Basics"
+permalink: /introduction/basics
+hide: true
+---
+
+A `GET` request can be performed by calling the `.get` class method:
+
+```ruby
+response = Faraday.get 'http://sushi.com/nigiri/sake.json'
+```
+
+This works if you don't need to set up anything; you can roll with the default middleware
+stack and default adapter (see [Faraday::RackBuilder#initialize][rack_builder]).
+
+## The Connection Object
+
+A more flexible way to use Faraday is to start with a Connection object. If you want to keep the same defaults, you can use this syntax:
+
+```ruby
+conn = Faraday.new(url: 'http://www.sushi.com')
+```
+
+Connections can also take an options hash as a parameter, or be configured with a block.
+Check out the [Middleware][middleware] page for more details about how to use this block for configurations.
+Since the default middleware stack uses the `url_encoded` middleware and default adapter, use them on building your own middleware stack.
+
+```ruby
+conn = Faraday.new(url: 'http://sushi.com') do |faraday|
+  # form-encode POST params
+  faraday.request :url_encoded
+  # log requests and responses to $stdout
+  faraday.response :logger
+  # make requests with Net::HTTP
+  faraday.adapter Faraday.default_adapter
+end
+```
+
+Once you have the connection object, use it to make HTTP requests. You can pass parameters to it in a few different ways:
+
+```ruby
+conn = Faraday.new(url: 'http://sushi.com/nigiri')
+
+## GET ##
+
+response = conn.get 'sake.json'
+# => GET http://sushi.com/nigiri/sake.json
+
+# Using an absolute path overrides the path from the connection initializer
+response = conn.get '/maki/platters.json'
+# => GET http://sushi.com/maki/platters.json
+ 
+# You can then access the response body
+response.body
+
+# Path can also be empty. Parameters can be provided as a hash.
+conn.get '', { name: 'Maguro' }
+# => GET http://sushi.com/nigiri?name=Maguro
+
+conn.get do |req|
+  req.url '/search', page: 2
+  req.params['limit'] = 100
+end
+# => GET http://sushi.com/search?limit=100&page=2
+
+## POST ##
+
+# Parameters for POST requests are automatically put in the body as
+# www-form-urlencoded.
+conn.post '', { name: 'Maguro' }
+# => POST "name=maguro" to http://sushi.com/nigiri
+
+# To post as JSON instead of www-form-urlencoded, set the request header
+conn.post do |req|
```

Comment:
Faraday doesn't automatically convert hashes to json without the `json` middleware in `faraday_middleware`. Doing this correctly looks like this:

```ruby
require 'faraday'
require 'faraday_middleware'
f = Faraday.new do |b|
  b.request :url_encoded
  b.request :json
  b.adapter Faraday.default_adapter
end

f.post $url, {data: "json"}, "Content-Type" => "application/json"
```

If the `json` middleware is not loaded, this is what you get:

```
>> Faraday.post $url, {ctype: "json"}, 'Content-Type' => 'application/json'
NoMethodError: undefined method `bytesize' for {:ctype=>"json"}:Hash
```

What do you think about moving this JSON bit to a page specifically for the `json` middleware? This section could then be clear that Faraday only handles url/form/multipart encoding for you.

I also tend to prefer handling JSON encoding explicitly:

```ruby
f.post $url, {data: "json"}.to_json, "Content-Type" => "application/json"
```

Reply:
Hey, yes I know the JSON middleware is not included in Faraday, but if you check my example it sets the body as a stringified JSON, so it should work out of the box.
This is of course reductive and cumbersome, but that's we have the JSON middleware in `faraday_middleware`, although I agree that deserves its own page (and arguably on another website?)

Reply:
Ahh, must've missed those quote, you got me :) I wonder if it should show generic examples of setting the req body to a string or io reader instead, even if the sample code is still sending raw json. What do you think?

1. Send form/multipart as ruby hash
2. Send string resp body with content type
3. Send io reader resp body with content type

Reply:
Totally makes sense 👍! Thanks